config/Dockerfiles/rpmbuild/rpmbuild-test.sh

```diff
@@ -1,79 +1,146 @@
-pushd ${OUTPUTDIR}/${fed_repo}_repo && createrepo .
+rpmlint ${RPMDIR}/ > ${LOGDIR}/rpmlint_out.txt
+pushd ${RPMDIR} && createrepo .
 popd
 # Run fedabipkgdiff against the newly created rpm
 rm -rf libabigail
-git clone git://sourceware.org/git/libabigail.git
+git clone -q git://sourceware.org/git/libabigail.git
 RPM_TO_CHECK=$(find /${fed_repo}/results_${fed_repo}/${VERSION}/*/ -name "${fed_repo}-${VERSION}*" | grep -v src)
-libabigail/tools/fedabipkgdiff --from ${ABIGAIL_BRANCH} ${RPM_TO_CHECK} &> ${OUTPUTDIR}/logs/fedabipkgdiff_out.txt
+libabigail/tools/fedabipkgdiff --from ${ABIGAIL_BRANCH} ${RPM_TO_CHECK} &> ${LOGDIR}/fedabipkgdiff_out.txt
 RPM_NAME=$(basename $RPM_TO_CHECK)
-echo "package_url=${HTTP_BASE}/${fed_branch}/repo/${fed_repo}_repo/$RPM_NAME" >> ${OUTPUTDIR}/logs/package_props.txt
+echo "package_url=${HTTP_BASE}/${fed_branch}/repo/${fed_repo}_repo/$RPM_NAME" >> ${LOGDIR}/package_props.txt
 echo "original_spec_nvr=${truenvr}" >> ${OUTPUTDIR}/logs/package_props.txt
 RPM_NAME=$(echo $RPM_NAME | rev | cut -d '.' -f 2- | rev)
 echo "nvr=${RPM_NAME}" >> ${OUTPUTDIR}/logs/package_props.txt
 
-exit 0
+if [ -z "${enable_rsync}" ]; then echo "Rsync not enabled. Exiting" ; exit 0 ; fi
+
+# If we do rsync, make sure we have the password
+if [ -z "${RSYNC_PASSWORD}" ]; then echo "Told to rsync but no RSYNC_PASSWORD env var" ; exit 1 ; fi
+# Perform rsync to artifacts.ci.centos.org
+if [ -z "${production}" ]; then
+    RSYNC_BRANCH=${RSYNC_BRANCH}/staging
+fi
+mkdir -p ${RSYNC_BRANCH}
+mkdir repo
+# Kill backgrounded jobs on exit
+function clean_up {
+    # Delete the rsync lock we placed
+     rsync -vr --delete $(mktemp -d)/ fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/
+}
+trap clean_up EXIT SIGHUP SIGINT SIGTERM
+# Write uuid to a lock file and store a backup
+uuidgen > file.lock
+cp file.lock uuid.saved
+while true; do
+    # Check if lock exists on remote server
+     while [[ $(rsync --ignore-existing --dry-run -avz file.lock fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir) != *"file.lock"* ]]; do
+          sleep 60
+     done
+     cp uuid.saved file.lock
+    # Push lock file with uuid to remote server
+     rsync --ignore-existing -avz file.lock fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/
+    # Pull lock file back
+     rsync -avz fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/file.lock file.lock
+    # If uuid matches, we can proceed
+     if [[ $(diff file.lock uuid.saved) == "" ]]; then
+          break
+     fi
+     sleep 60
+done
+# Rsync the empty directories over first, then the repo directory
+rsync -arv ${RSYNC_BRANCH}/ fedora-atomic@artifacts.ci.centos.org::fedora-atomic
+if [ -z "${production}" ]; then
+     rsync --delete --stats -a ${RPMDIR} fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}
```

Comment:
The intent was that yes, because there was discussion of for the staging branch, to just keep one directory of rpms in the same place for just one package at a time, so it would clear the old rpms first.  However, I have updated the container now to use RSYNC_USER, RSYNC_SERVER and RSYNC_DIR, so everything should be much cleaner now.